ruby/ql/lib/codeql/ruby/dataflow/internal/DataFlowDispatch.qll

```diff
@@ -674,6 +721,11 @@ private predicate singletonMethodOnModule(MethodBase method, string name, Module
   )
   or
   flowsToSingletonMethodObject(trackModuleAccess(m), method, name)
+  or
+  exists(Module other |
+    extendCallModule(m, other) and
+    method = lookupMethod(other, name)
```

Comment:
Should this be limited to public methods?

Reply:
Good point. `private` can be called, but only if the receiver is `self` and `protected` methods can be called if the receiver is a subclass of the enclosing class or something.